src/docbuild/models/config_model/app.py

```diff
@@ -1,79 +1,140 @@
+    model_config = ConfigDict(extra='allow', populate_by_name=True) 
+
+    format: str | None = None
+    datefmt: str | None = None
+    style: Literal['%', '{', '$'] | None = None
+    class_name: str | None = Field(None, alias='class') 
+    validate: bool | None = None
+
+
+class HandlerConfig(BaseModel):
+    """
+    Defines the configuration for a single logging handler.
+    Allows extra keys specific to certain handler classes (e.g., 'filename', 'maxBytes').
+    """
+    model_config = ConfigDict(extra='allow', populate_by_name=True) 
+
+    class_name: str | None = Field(None, alias='class')
+    level: str | int | None = None
+    formatter: str | None = Field(None, description="Must match a key in 'formatters'.")
+    filters: list[str] | None = None
+
+
+class LoggerConfig(BaseModel):
+    """
+    Defines the configuration for a single logger instance.
+    """
+    model_config = ConfigDict(extra='allow', populate_by_name=True)
+
+    level: str | int | None = None
+    handlers: list[str] | None = Field(None, description="Must match keys in 'handlers'.")
+    propagate: bool | None = True
+    filters: list[str] | None = None
+
+
+class RootLoggerConfig(BaseModel):
+    """
+    Defines the configuration for the root logger.
+    """
+    model_config = ConfigDict(extra='allow', populate_by_name=True)
     
-    This structure is validated for existence of core sections, but allows
-    for the flexible structure required by logging.config.dictConfig.
+    # Provide a default level and handlers list to prevent crashes 
+    # when logging.config.dictConfig is called without them.
+    level: str | int | None = Field('WARNING', description="The minimum severity level to log.")
+    handlers: list[str] | None = Field(default_factory=list, description="Must match keys in 'handlers'.")
+    filters: list[str] | None = None
+
+# --- 2. Type Aliases for Dictionaries ---
+
+type FormattersDict = dict[str, FormatterConfig]
+type HandlersDict = dict[str, HandlerConfig]
+type LoggersDict = dict[str, LoggerConfig]
+type FiltersDict = dict[str, Any]
+
+# --- 3. The Top-Level Logging Configuration Model (Replaces App_LoggingConfig) ---
+
+class App_LoggingConfig(BaseModel):
     """
-    version: int = 1
+    The complete Pydantic model for Python's logging.dictConfig.
+    """
+    version: Literal[1] = Field(description="The schema version. Must be 1.")
     disable_existing_loggers: bool = False
+
+    formatters: FormattersDict = Field(default_factory=dict, description="All configured formatters.")
+    filters: FiltersDict = Field(default_factory=dict, description="All configured filters.")
+    handlers: HandlersDict = Field(default_factory=dict, description="All configured handlers.")
+    loggers: LoggersDict = Field(default_factory=dict, description="All specific loggers (e.g., 'my_app').")
+    root: RootLoggerConfig = Field(default_factory=RootLoggerConfig, description="The root logger configuration.")
```

Comment:
I stumbled upon why `default_factory` is dict for the others and `RootLoggerConfig` for root. I think we should add a comment:


```suggestion
    # Use an instance as we have only one root logger
    root: RootLoggerConfig = Field(default_factory=RootLoggerConfig, description="The root logger configuration.")
```